go.mod

```diff
@@ -15,14 +15,17 @@ require (
 	golang.org/x/tools v0.1.6-0.20210820212750-d4cc65f0b2ff
```

Comment:
This scenario does not solve the case: https://github.com/kubernetes-sigs/kubebuilder/pull/2581

Reply:
With this version, it does work with my environment with golang 1.17 + mac.
Can you please elaborate?

Reply:
The problem faced/reported on the issue https://github.com/kubernetes-sigs/kubebuilder/pull/2581 is not solved with this PR.

Reply:
Did you try it?
Because preventing the error provides a non nil Runtime(which should fix the bug)

Reply:
PS that is why I added the error returning (which you asked me to revert)

Reply:
Yes, I tried it locally. 
The issue raised in the issue  https://github.com/kubernetes-sigs/kubebuilder/issues/2567 is not solved with the changes in this PR. See; https://github.com/kubernetes-sigs/kubebuilder/issues/2567#issuecomment-1086630767

Reply:
Ok, my bad :)

Thanks for correcting me